java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -429,25 +429,29 @@ private boolean shouldTerminate(Set<LearnedConstraint> learnedConstraints) {
     }
 
     private Flow flowFromLearnedAssociation(LearnedAssociation learnedAssociation, ExplodedGraph.Node node) {
-      Flow.Builder flowBuilder = Flow.builder();
       ProgramState programState = node.programState;
       Preconditions.checkState(programState != null, "Learned association with null state in parent node of the edge.");
       Symbol rhsSymbol = symbolFromStack(learnedAssociation.symbolicValue(), programState);
       if (rhsSymbol != null) {
-        flowBuilder.add(location(node, String.format(IMPLIES_SAME_VALUE, learnedAssociation.symbol().name(), rhsSymbol.name())));
-      } else {
-        ConstraintsByDomain allConstraints = programState.getConstraints(learnedAssociation.symbolicValue());
-        Collection<Constraint> constraints = filterByDomains(allConstraints);
-        for (Constraint constraint : constraints) {
-          String symbolName = learnedAssociation.symbol().name();
-          String msg;
-          if (assigningNullFromTernary(node) || (assigningFromMethodInvocation(node) && assignedFromYieldWithUncertainResult(constraint, node))) {
-            msg = IMPLIES_CAN_BE_MSG;
-          } else {
-            msg = IMPLIES_IS_MSG;
-          }
-          flowBuilder.add(location(node, String.format(msg, symbolName, constraint.valueAsString())));
+        return Flow.of(location(node, String.format(IMPLIES_SAME_VALUE, learnedAssociation.symbol().name(), rhsSymbol.name())));
+      }
+      Flow.Builder flowBuilder = Flow.builder();
+      ConstraintsByDomain allConstraints = programState.getConstraints(learnedAssociation.symbolicValue());
+      Collection<Constraint> constraints = filterByDomains(allConstraints);
+      boolean isBoolean = learnedAssociation.symbol.type().is("boolean");
```

Comment:
I think I would prefer here to test `type().isPrimitive(Primitives.BOOLEAN)`, or even  `type().isPrimitive()` , as none of the primitive type can be null.